aiida/orm/nodes/data/array/bands.py

```diff
@@ -1074,6 +985,117 @@ def show_mpl(self, **kwargs):
+
+        if prettify_format is None:
+            # Default. Specified like this to allow caller functions to pass 'None'
+            prettify_format = 'gnuplot_seekpath'
+
+        plot_info = self._get_bandplot_data(
+            cartesian=True, prettify_format=prettify_format, join_symbol='|', y_origin=y_origin)
+
+        bands = plot_info['y']
+        x = plot_info['x']
+        labels = plot_info['labels']
+
+        num_labels = len(labels)
+        num_bands = bands.shape[1]
+
+        # axis limits
+        if y_max_lim is None:
+            y_max_lim = bands.max()
+        if y_min_lim is None:
+            y_min_lim = bands.min()
+        x_min_lim = min(x)  # this isn't a numpy array, but a list
+        x_max_lim = max(x)
+
+        # first prepare the xy coordinates of the sets
+        raw_data, _ = self._prepare_dat_blocks(plot_info, comments=comments)
+
+        xtics_string = ', '.join('"{}" {}'.format(label, pos) for pos, label in plot_info['labels'])
+
+        script = []
+        # Start with some useful comments
+
+        if comments:
+            script.append(prepare_header_comment(self.uuid, plot_info=plot_info, comment_char='# '))
+        script.append('')
+
+        script.append(u"""## Uncomment the next two lines to write directly to PDF
+## Note: You need to have gnuplot installed with pdfcairo support!
+#set term pdfcairo
+#set output 'out.pdf'
+
+### Uncomment one of the options below to change font
+### For the LaTeX fonts, you can download them from here:
+### https://sourceforge.net/projects/cm-unicode/
+### And then install them in your system
+## LaTeX Serif font, if installed
+#set termopt font "CMU Serif, 12"
+## LaTeX Sans Serif font, if installed
+#set termopt font "CMU Sans Serif, 12"
+## Classical Times New Roman
+#set termopt font "Times New Roman, 12"
+""")
+
+        # Actual logic
+        script.append('set termopt enhanced')  # Properly deals with e.g. subscripts
+        script.append('set encoding utf8')  # To deal with Greek letters
+        script.append('set xtics ({})'.format(xtics_string))
+
+        script.append('unset key')
+
+
+        script.append('set yrange [{}:{}]'.format(y_min_lim, y_max_lim))
+
+        script.append('set ylabel "{}"'.format('Dispersion ({})'.format(self.units)))
+
+        if title:
+            script.append('set title "{}"'.format(title.replace('"', '\"')))
+
+        # Plot, escaping filename
+        if len(x) > 1:
+            script.append('set xrange [{}:{}]'.format(x_min_lim, x_max_lim))
+            script.append('set grid xtics lt 1 lc rgb "#888888"')
+            script.append('plot "{}" with l lc rgb "#000000"'.format(os.path.basename(dat_filename).replace('"', '\"')))
+        else:
+            script.append('set xrange [-1.0:1.0]')
+            script.append('plot "{}" using ($1-0.25):($2):(0.5):(0) with vectors nohead lc rgb "#000000"'.format(os.path.basename(dat_filename).replace('"', '\"')))
```

Comment:
Another place I modified in this function. In order to handle single kpoint situation.